src/components/FilterBar/FilterDisplay/Inputs/SelectInput.react.js

```diff
@@ -0,0 +1,36 @@
+export class SelectInput extends React.Component {
+  constructor(props) {
+    super(props)
+    var filter = props.filterBarActor.getFilter(props.filterUid);
+
+    this.state = { value: (filter.value || filter.options[0].value) };
+    this.props.filterBarActor.updateFilter(this.props.filterUid, 'value', this.state.value);
+  }
+
+  _onChange(event) {
+    this.setState({value: event.target.value});
```

Comment:
If the change isn't being listened to elsewhere you might to to listen to the change in this component and any components that are using the same pattern